### PR TITLE
Feature/faster build

### PR DESCRIPTION
**Pull request by khancyr**
correct some python code on update.py
convert wget call to pure python equivalent that are faster. Make the download files from https://autotest.ardupilot.org in parallele to speed up the build

### CODE REVIEW

File: common_conf.py
Comment (Hwurzburg) [inline common_conf.py:63]: is this now undefined?

**Comment by Hwurzburg** [comment]
Seems a lot faster on my local builds....even without the --fast option

File: update.py
Comment (TunaLobster) [inline update.py:531]: Maybe update to f-strings while doing this?

Comment (khancyr) [reply update.py]: done

File: common_conf.py
Comment (khancyr) [reply common_conf.py:63]: it is still present L19

File: scripts/cap_params.py
Comment (peterbarker) [inline scripts/cap_params.py:24]: Was this change tested?

File: update.py
Comment (peterbarker) [inline update.py:140]: This yieldssyntax isn't in *all* of Python3, is it?  When did it come in?

File: update.py
Comment (peterbarker) [inline update.py:148]: Why should we continue if this fails?!

File: update.py
Comment (peterbarker) [inline update.py:162]: Isn't this just going to die later?

File: update.py
Comment (peterbarker) [inline update.py:204]: These types make things rather less readable.

**Review by peterbarker — CHANGES_REQUESTED** [review]
I think I might quite like the meat of these changes.

But they are mixed in with a whole heap of changes to update the Python code.  Most of which are good - unicode string stuff, utf-declarations, fstrings etc.  This just makes it far more difficult to review than it should be.

This is only a partial review as I got frustrated with working through changes unrelated to the changes in the parameter and log information download.

**Comment by peterbarker** [comment]
> @peterbarker I didn't intend to make a huge python3 style update, but Charlie said : hey why not use f-string. So I ran pyupgrade to modernize all .py in one go.
> 
> Beside the file downloading, I didn't intend to modify the logic nor refactor everything.

You could probably do that style update and rebase on top of that, then.

I'm still a touch worried about (a) the "yields" syntax which I thought was relatively new, and the readability with the Types system.   OTOH, I was tired and a bit grumpy on first review, so we'll see what happens on a second review :-)

Love f-strings :-)

**Comment by TunaLobster** [comment]
Sphinx 5.1.1 is the current minimum for the ArduPilot wiki. That requires python 3.6 or newer. Generators showed up before then so everything is good there.

Type hints are also great for quickly coming up to speed with what is happening. They don't do much until coupled with a linter or static analyzer.

**Comment by peterbarker** [comment]
> Sphinx 5.1.1 is the current minimum for the ArduPilot wiki. That requires python 3.6 or newer. Generators showed up before then so everything is good there.

Ah, sorry, I was referring to what's apparently known as "return type annotations".  I was using the older meaning of "yields" as synonymous with "returns", as opposed to the Pythonesque "yields" for generators.

> Type hints are also great for quickly coming up to speed with what is happening. They don't do much until coupled with a linter or static analyzer.

Are they, 'though?  It's harder to just glance at a function definition and see what it's taking.  I do think we should find out - but introducing their syntax really should be done in a separate PR.  If we can show that flake8 can now pick up more bugs, bringing it in would be a no-brainer IMO.  I don't think one could argue that it does make the syntax more scary, 'though.

Please note that you don't need my approval to merge this PR.  If you want to merge it with Henry's blessing that's workable.  It's just that if it breaks its likely to stay broken until you fix it :-)

We're not running out of PR numbers.  I would have merged the unicode and f-string changes instantly.

**Comment by khancyr** [comment]
@peterbarker flake8 doesn't do type checking. The best tool for now is mypy or IDE if it support typing. 
One issue thus is that without setting at least the return type, most analyzers will just consider that the function take anything as input and return anything so they only report big error. Using typing, you have a better check on the return condition and avoid some strange casting.
I am not totally fine with typing as is could be really annoying to type and read, but it provides good bug tracking ... so I try to add it by default now

**Comment by Hwurzburg** [comment]
@peterbarker Peter, still needs changes? or can I merge?

**Comment by Hwurzburg** [comment]
@khancyr  @peterbarker says his review stands....I wont merge unless you two come to an understanding...sorry

**Comment by khancyr** [comment]
no problem, I will improve this

**Comment by khancyr** [comment]
@peterbarker removed py3 upgrade change, just the speedup now, will make an other PR for py3

**Pull request merged**
